packages/compose/src/hooks/use-focus-outside/test/index.js

```diff
@@ -4,120 +4,149 @@
+		{ /* Wrapper */ }
 		<div { ...useFocusOutside( callback ) }>
 			<input type="text" />
-			<input type="button" />
+			<button>Button inside the wrapper</button>
+			<iframe title="test-iframe">
+				<button>Inside the iframe</button>
+			</iframe>
 		</div>
-	);
 
-	class TestComponent extends Component {
-		render() {
-			return <FocusOutsideComponent { ...this.props } />;
-		}
-	}
+		<button>Button outside the wrapper</button>
+	</div>
+);
 
-	beforeEach( () => {
-		// Mock document.hasFocus() to always be true for testing
-		// note: we overide this for some tests.
-		origHasFocus = document.hasFocus;
-		document.hasFocus = () => true;
-
-		onFocusOutside = jest.fn();
-	} );
-
-	afterEach( () => {
-		document.hasFocus = origHasFocus;
-	} );
-
-	it( 'should not call handler if focus shifts to element within component', () => {
-		render( <TestComponent onFocusOutside={ onFocusOutside } /> );
+describe( 'useFocusOutside', () => {
+	it( 'should not call handler if focus shifts to element within component', async () => {
+		const mockOnFocusOutside = jest.fn();
+		const user = userEvent.setup( {
+			advanceTimers: jest.advanceTimersByTime,
+		} );
 
-		const input = screen.getByRole( 'textbox' );
-		const button = screen.getByRole( 'button' );
+		render(
+			<FocusOutsideComponent onFocusOutside={ mockOnFocusOutside } />
+		);
 
-		input.focus();
-		input.blur();
-		button.focus();
+		// Tab through the interactive elements inside the wrapper,
+		// causing multiple focus/blur events.
+		await user.tab();
+		expect( screen.getByRole( 'textbox' ) ).toHaveFocus();
 
-		jest.runAllTimers();
+		await user.tab();
+		expect(
+			screen.getByRole( 'button', { name: 'Button inside the wrapper' } )
+		).toHaveFocus();
 
-		expect( onFocusOutside ).not.toHaveBeenCalled();
+		expect( mockOnFocusOutside ).not.toHaveBeenCalled();
 	} );
 
 	it( 'should not call handler if focus transitions via click to button', async () => {
+		const mockOnFocusOutside = jest.fn();
 		const user = userEvent.setup( {
 			advanceTimers: jest.advanceTimersByTime,
 		} );
-		render( <TestComponent onFocusOutside={ onFocusOutside } /> );
-
-		const input = screen.getByRole( 'textbox' );
-		const button = screen.getByRole( 'button' );
 
-		input.focus();
-		await user.click( button );
+		render(
+			<FocusOutsideComponent onFocusOutside={ mockOnFocusOutside } />
+		);
 
-		jest.runAllTimers();
+		// Click the input and the button, causing multiple focus/blur events.
+		await user.click( screen.getByRole( 'textbox' ) );
+		await user.click(
+			screen.getByRole( 'button', { name: 'Button inside the wrapper' } )
+		);
 
-		expect( onFocusOutside ).not.toHaveBeenCalled();
+		expect( mockOnFocusOutside ).not.toHaveBeenCalled();
 	} );
 
-	it( 'should call handler if focus doesn’t shift to element within component', () => {
-		render( <TestComponent onFocusOutside={ onFocusOutside } /> );
+	it( "should call handler if focus doesn't shift to element within component", async () => {
```

Comment:
I think this could read easier as something like:
```suggestion
	it( "should call handler if focus shifts to element outside component", async () => {
```

Reply:
Good point! Done in https://github.com/WordPress/gutenberg/pull/45317/commits/ba97cff2d5d163304ff46d3b1faa32843b106bb2